storage-retrieval/lib/src/client.rs

```diff
@@ -0,0 +1,252 @@
+    ) -> Result<(), ClientError> {
+        trace!(?event, "Received swarm event");
+
+        match event {
+            SwarmEvent::ConnectionEstablished {
+                peer_id,
+                connection_id,
+                endpoint,
+                ..
+            } => {
+                self.on_peer_connected(peer_id, connection_id, endpoint);
+            }
+            SwarmEvent::ConnectionClosed {
+                peer_id,
+                connection_id,
+                ..
+            } => {
+                self.on_peer_disconnected(peer_id, connection_id);
+            }
+            SwarmEvent::Behaviour(BehaviourEvent::Bitswap(event)) => {
+                self.on_bitswap_event(event).await?;
+            }
+            _ => {
+                // Nothing to do here
+            }
+        }
+
+        Ok(())
+    }
+
+    #[instrument(skip_all, fields(peer_id = %peer_id))]
+    fn on_peer_connected(
+        &mut self,
+        peer_id: PeerId,
+        _connection_id: ConnectionId,
+        _endpoint: ConnectedPoint,
+    ) {
+        debug!("Peer connected");
+    }
+
+    #[instrument(skip_all, fields(peer_id = %peer_id))]
+    fn on_peer_disconnected(&mut self, peer_id: PeerId, _connection_id: ConnectionId) {
+        debug!("Peer disconnected");
+    }
+
+    #[instrument(level = "trace", skip(self))]
+    async fn on_bitswap_event(&mut self, event: beetswap::Event) -> Result<(), ClientError> {
+        match event {
+            beetswap::Event::GetQueryResponse { query_id, data } => {
+                let Some(cid) = self.queries.remove(&query_id) else {
+                    return Ok(());
+                };
+
+                // Store the received block to a blockstore
+                self.blockstore.put_keyed(&cid, &data).await?;
+                info!("received new block {cid:?}");
+
+                match cid.codec() {
+                    DAG_PB_CODE => {
+                        let node = <DagPbCodec as Codec<PbNode>>::decode_from_slice(&data).unwrap();
+
+                        // Request a block for each new discoverd link
+                        node.links
+                            .iter()
+                            .for_each(|link| self.request_block(link.cid));
+                    }
+                    RAW_CODE => {
+                        debug!("{cid} raw block. nothing to do");
+                    }
+                    _other => {
+                        error!("{cid} codec {_other} not supported");
+                    }
+                }
+            }
+            beetswap::Event::GetQueryError { query_id, error } => {
+                if let Some(cid) = self.queries.remove(&query_id) {
+                    info!("received error for {cid:?}: {error}");
+                }
+            }
+        }
+
+        Ok(())
+    }
+}
+
+/// The blockstore used by the client. It simulates a blockstore that never
+/// holds any blocks.
+struct EmptyBlockstore;
```

Comment:
Empty works but I feel that `Passthrough` is more adequate since it also better explains its goal
```suggestion
struct PassthroughBlockstore;
```

Reply:
YES. I couldn't find a better name. Empty didn't sit quite right.